BridgeCareApp/VuejsApp/src/services/scenario.service.ts

```diff
@@ -21,6 +21,50 @@ export default class ScenarioService {
         });
     }
 
+    static getLegacyScenarios(scenarios: Scenario[]): AxiosPromise {
+        return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
+
+            axiosInstance.get('api/simulations')
+                .then((responseLegacy: AxiosResponse<Scenario[]>) => {
+                    if (hasValue(responseLegacy)) {
+                        var resultant: Scenario[] = [];
+                        responseLegacy.data.forEach(simulation => {
+
+                            if (hasValue(scenarios)) {
+                                var isPresent = false;
+                                scenarios.forEach(scenario => {
```

Comment:
You can use some ramda functions here instead of doing an inner loop. Look at the any & propEq functions. There are examples in code. It would look something like..if (any(propEq('simulationId', simulation.simulationId), scenarios)) {...}. You can even use responseLegacy.filter in combination with any & propEq instead of doing the loops.

Reply:
Made the changes